SYCL/Reduction/reduction_usm.cpp

```diff
@@ -16,13 +16,13 @@
 
 using namespace sycl;
 
-template <typename T1, typename T2> class KName;
+template <typename T1, typename T2> class USMKName;
```

Comment:
This code/test was written when naming lambda functions was required.
If you want to make the code even simpler, then this class can be just removed.

Reply:
I thought about removing all these names, but then I'd still need to have a test for the cases when unnamed lambda are disabled.

As such, not going to implement as part of this PR, maybe in future.